source/common/http/filter_manager.cc

```diff
@@ -306,11 +321,19 @@ void ActiveStreamFilterBase::traversePerFilterConfig(
         cb(config);
       });
 
-  if (handled || filter_context_.filter_name == filter_context_.config_name) {
+  if (handled || parent_.no_downgrade_to_canonical_name_ ||
+      filter_context_.filter_name == filter_context_.config_name) {
     return;
   }
 
-  current_route->traversePerFilterConfig(filter_context_.filter_name, cb);
+  current_route->traversePerFilterConfig(
+      filter_context_.filter_name, [&cb](const Router::RouteSpecificFilterConfig& config) {
+        ENVOY_LOG_FIRST_N(warn, 10,
```

Comment:
Let's create a tracking issue for this, and link to it.  If folks have a problem either with the warning, or with the behavior change, it'd be good for them to know where to go.

Reply:
got it. will create a issue and add the link to the warning log.